### PR TITLE
Add Beta support for auto_healing_policies to instance group manager.

### LINKED ISSUE #48: provider/google: Support for managed instance group autohealing
_This issue was originally opened by @premist as hashicorp/terraform#10360. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Managed instance group on Google Compute Engine has a feature where you can monitor and automatically re-create instances based on associated health check. You can use `instanceGroupManagers.setAutoHealingPolicies` to apply the health check to the managed instance group, like the following:

```http
POST https://www.googleapis.com/compute/beta/projects/[PROJECT_ID]/zones/[ZONE]/instanceGroupManagers/[INSTANCE_GROUP]/setAutoHealingPolicies

{
 "autoHealingPolicies": [
  {
   "healthCheck": "global/httpHealthChecks/example-check",
   "initialDelaySec": 120
  }
 ]
}
```

Note that this API is currently in beta.

### Documentations

- [API Reference](https://cloud.google.com/compute/docs/reference/beta/instanceGroupManagers/setAutoHealingPolicies)
- [Setting up health checking for managed instance groups](https://cloud.google.com/compute/docs/instance-groups/creating-groups-of-managed-instances#monitoring_groups)

### PR DESCRIPTION
**Pull request by rileykarson**
Add support for the first Beta feature, `auto_healing_policies`, to `google_compute_instance_group_manager`. Beta support was added to the resource in #234.

Part of #93.
Fixes #48.

Beta features are documented by adding a third divider to the list of fields; in addition to `Required`, and `Optional`, there is now a `Beta` section, where Beta fields are listed.

When users begin using Beta fields, their resources will automatically "upgrade" to Beta, and begin performing operations using Beta APIs instead of v1. If they remove the Beta fields from their resource, it will then automatically "upgrade" back to v1.

By inferring the version from the fields a user uses, we;
- Don't need to perform validation that we match different schema versions in CRUD methods.
- Preserve existing Terraform state files.
- Allow seamless "upgrades" back to v1 as features enter GA.

We do not currently have a means of explicitly notifying users what version an operation is being performed at;  I'd like to investigate means of how to show the version to users and especially whether we can surface that information at the plan phase before committing to an implementation. 

I don't believe that it should be blocking for this change, as both our documentation and the GCP documentation will communicate that a feature is in Beta to users.

@catsby

### CODE REVIEW

File: google/resource_compute_instance_group_manager_test.go [hunk 1]
```diff
@@ -704,6 +799,121 @@ func testAccInstanceGroupManager_separateRegions(igm1, igm2 string) string {
 	`, igm1, igm2)
 }
 
+func testAccInstanceGroupManager_autoHealingPolicies(template, target, igm, hck string) string {
+	return fmt.Sprintf(`
+	resource "google_compute_instance_template" "igm-basic" {
```
Comment (danawillow) [inline google/resource_compute_instance_group_manager_test.go]: Can you align this to the left?

Comment (rileykarson) [inline google/resource_compute_instance_group_manager_test.go]: Done.

**Comment by danawillow** [comment]
Thanks! I'm fine with this once @catsby is.

File: website/docs/r/compute_instance_group_manager.html.markdown [hunk 2]
```diff
@@ -81,12 +81,24 @@ The following arguments are supported:
     instances in the group are added. Updating the target pools attribute does
     not affect existing instances.
 
+---
+
+* `auto_healing_policies` - (Optional, Beta) 	The autohealing policies for this managed instance
```
Comment (rosbo) [inline website/docs/r/compute_instance_group_manager.html.markdown]: nit: extra spaces between `(Optional, Beta)` and `The`

File: google/resource_compute_instance_group_manager_test.go [hunk 3]
```diff
@@ -704,6 +799,127 @@ func testAccInstanceGroupManager_separateRegions(igm1, igm2 string) string {
...
+	request_path       = "/"
+	check_interval_sec = 1
+	timeout_sec        = 1
+}
+	`, template, target, igm, hck)
+}
+
+// This test is to make sure that a single version resource can link to a versioned resource
+// without perpetual diffs because the self links mismatch.
+// Once auto_healing_policies is no longer beta, we will need to use a new field or resource.
```
Comment (rosbo) [inline google/resource_compute_instance_group_manager_test.go]: a new field or resource for what? Not super clear.

File: google/resource_compute_instance_group_manager_test.go [hunk 4]
```diff
@@ -185,6 +186,71 @@ func TestAccInstanceGroupManager_separateRegions(t *testing.T) {
...
+	hck := fmt.Sprintf("igm-test-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckInstanceGroupManagerDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccInstanceGroupManager_autoHealingPolicies(template, target, igm, hck),
+				Check: resource.ComposeTestCheckFunc(
```
Comment (rosbo) [inline google/resource_compute_instance_group_manager_test.go]: No need to use `ComputeTestCheckFunc` if you only have one check

File: google/resource_compute_instance_group_manager_test.go [hunk 5]
```diff
@@ -185,6 +186,71 @@ func TestAccInstanceGroupManager_separateRegions(t *testing.T) {
...
+					testAccCheckInstanceGroupManagerBetaExists(
+						"google_compute_instance_group_manager.igm-basic", &manager),
+				),
+			},
+		},
+	})
+
+	if len(manager.AutoHealingPolicies) != 1 {
```
Comment (rosbo) [inline google/resource_compute_instance_group_manager_test.go]: Have you considered moving these in a `TestCheckFunc`?

File: website/docs/r/compute_instance_group_manager.html.markdown (hunk 2 above)
Comment (rileykarson) [inline website/docs/r/compute_instance_group_manager.html.markdown]: Done.

File: google/resource_compute_instance_group_manager_test.go (hunk 4 above)
Comment (rileykarson) [inline google/resource_compute_instance_group_manager_test.go]: Done.

File: google/resource_compute_instance_group_manager_test.go (hunk 5 above)
Comment (rileykarson) [inline google/resource_compute_instance_group_manager_test.go]: Done.

File: google/resource_compute_instance_group_manager_test.go (hunk 3 above)
Comment (rileykarson) [inline google/resource_compute_instance_group_manager_test.go]: Done.

**Comment by rileykarson** [comment]
@catsby: That sounds good, I'll submit the header and update this page to link to it as a followup PR. 

```
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./google -v -run=TestAccInstanceGroupManager -timeout 120m
=== RUN   TestAccInstanceGroupManager_importBasic
--- PASS: TestAccInstanceGroupManager_importBasic (295.09s)
=== RUN   TestAccInstanceGroupManager_importUpdate
--- PASS: TestAccInstanceGroupManager_importUpdate (90.95s)
=== RUN   TestAccInstanceGroupManager_basic
--- PASS: TestAccInstanceGroupManager_basic (253.15s)
=== RUN   TestAccInstanceGroupManager_targetSizeZero
--- PASS: TestAccInstanceGroupManager_targetSizeZero (44.80s)
=== RUN   TestAccInstanceGroupManager_update
--- PASS: TestAccInstanceGroupManager_update (156.22s)
=== RUN   TestAccInstanceGroupManager_updateLifecycle
--- PASS: TestAccInstanceGroupManager_updateLifecycle (104.03s)
=== RUN   TestAccInstanceGroupManager_updateStrategy
--- PASS: TestAccInstanceGroupManager_updateStrategy (210.90s)
=== RUN   TestAccInstanceGroupManager_separateRegions
--- PASS: TestAccInstanceGroupManager_separateRegions (213.87s)
=== RUN   TestAccInstanceGroupManager_autoHealingPolicies
--- PASS: TestAccInstanceGroupManager_autoHealingPolicies (91.30s)
=== RUN   TestAccInstanceGroupManager_selfLinkStability
--- PASS: TestAccInstanceGroupManager_selfLinkStability (123.06s)
PASS
ok  	github.com/terraform-providers/terraform-provider-google/google	1583.506s
```

**Pull request merged**